### PR TITLE
include scheduler_on_plateau in optim.h

### LINKED ISSUE #121593: reduce_on_plateau_scheduler.h missing from optim.h in libtorch ?
### 🚀 The feature, motivation and pitch

Hello,

I have been able to successfully use ReduceLROnPlateauScheduler in C++ with libtorch, but I had to include it manually:

`#include <torch/optim/schedulers/reduce_on_plateau_scheduler.h>`

Maybe you forgot to include it in the file:

`torch/csrc/api/include/torch/optim.h`

Or it is intentional ?

### Alternatives

_No response_

### Additional context

_No response_

cc @jbschlosser @vincentqb @albanD @janeyx99 @crcrpar

### PR DESCRIPTION
**Pull request by blegouix**
Fixes #121593

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
can you add a unit test similar to https://github.com/pytorch/pytorch/blob/a157a0d00de39fa15f1fd7a5c7c49865bac04f01/test/cpp/api/optim.cpp#L513?

**Comment by blegouix** [comment]
done but not tested, I don't have a local install of pytorch

File: test/cpp/api/optim.cpp [hunk 1]
```diff
@@ -523,3 +523,18 @@ TEST(OptimTest, CheckLRChange_StepLR_Adam) {
 
   check_lr_change(optimizer, step_lr_scheduler, expected_epoch_lrs);
 }
+
+TEST(OptimTest, CheckLRChange_ReduceLROnPlateau_Adam) {
+  torch::Tensor parameters = torch::zeros({1});
+  auto optimizer = Adam({parameters}, AdamOptions().lr(1e-3));
+
+	const SchedulerMode scheduler_mode = SchedulerMode::min;
```
Comment (janeyx99) [inline test/cpp/api/optim.cpp:531]: nit: is the irregular spacing here intentional?

Comment (blegouix) [reply test/cpp/api/optim.cpp:531]: No, can you finish the PR ? At first I just wanted to warn you that an include was missing, I do not want to do indentation, verify test pass etc.

Comment (janeyx99) [reply test/cpp/api/optim.cpp:531]: Ah, I see haha, thanks for the honesty. We'll have someone else pick up the work here--thanks for the warning!

**Review by albanD — APPROVED** [review]
Ok!

File: test/cpp/api/optim.cpp (hunk 1 above)
Comment (blegouix) [reply test/cpp/api/optim.cpp]: Thanks for understanding :)

**Pull request closed without merging**